hostfactory/host_provider/src/cluster.py

```diff
@@ -13,6 +13,37 @@
 except ImportError:
     import cyclecliwrapper as cyclecli
 
+class AutoscalingStrategy(enumerate):
+    PRICE = "price"
+    CAPACITY = "capacity"
+    WEIGHTED = "weighted"
+
+# This is only for unit testing
+def allocation_strategy(node_mgr, template_id, slot_count, capacity_limit_timeout, vm_sizes):
```

Comment:
Refactor this standalone algorithm so that it takes in weights and performs a common allocation algorithm using that.

Break out the weight calculations so that they are easier to test - i.e. separation of responsibilities.